src/dotnet/commands/dotnet-install/dotnet-install-tool/InstallToolCommand.cs

```diff
@@ -55,21 +55,28 @@ internal class InstallToolCommand : CommandBase
             _verbosity = appliedCommand.SingleArgumentOrDefault("verbosity");
 
             var cliFolderPathCalculator = new CliFolderPathCalculator();
-            var offlineFeedPath = new DirectoryPath(cliFolderPathCalculator.CliFallbackFolderPath);
-            _toolPackageObtainer = toolPackageObtainer ?? new ToolPackageObtainer(
-                                       new DirectoryPath(cliFolderPathCalculator.ToolsPackagePath),
-                                       offlineFeedPath,
-                                       () => new DirectoryPath(Path.GetTempPath())
-                                           .WithSubDirectories(Path.GetRandomFileName())
-                                           .WithFile(Path.GetRandomFileName() + ".csproj"),
-                                       new Lazy<string>(BundledTargetFramework.GetTargetFrameworkMoniker),
-                                       new ProjectRestorer(reporter));
+
+            FilePath? configFile = null;
+            if (_configFilePath != null)
+            {
+                configFile = new FilePath(_configFilePath);
+            }
+
+            _toolPackageInstaller = toolPackageInstaller
+                ?? new ToolPackageInstaller(
+                    new ToolPackageRepository(
+                        new DirectoryPath(cliFolderPathCalculator.ToolsPackagePath)),
+                    new ProjectRestorer(
```

Comment:
ProjectRestorer is a pretty leaky abstraction. And ToolPackageInstaller should be the only one "config" it. 

Part of the logic passed to nuget is via temp project. And another set of it are from params to ProjectRestorer. It is unfortunate, because of no clean API available. However, at least we can hide it under ToolPackageInstaller.

Reply:
Agreed that this is pretty leaky.

I figured that the `source`, `verbosity`, and `nugetConfig` parameters to install where really behavioral properties of the "restorer" and not at all important to the installer because they were being directly passed through to the call to restore the project.  From my perspective, the installer just wants to "restore" the temp project it creates; it doesn't care how that's done and it's up to how the restorer is configured that determines its behavior.

If you think these still belong as parameters to `InstallPackage`, I'll put them back.

Reply:
`source`
`nugetconfig`

might be a group to local the feed

But we also have additional feed (in fallbackfolder) which has to be in `InstallPackage`

verbosity is another group

`rid`, `tfm` should be more close to nugetconfig and source. But `rid` is in restorer and `tfm` is in temp project.

i don't feel there is a good line to draw. I feel just hide it under InstallPackage would be better. Maybe create a parameter object?

btw how did you refactor the test? all tests are start with InstallPackage, there is no setup involve restorer

Reply:
With respect to the tests, `Setup` in the installer tests creates a mock restorer if using the mock objects.

The mock restorer reads from a fake project file passed to `Restore` that tells it which package is to be restored (including version and TFM, similar to a package reference and various NuGet-related build properties).  The mock restorer then simulates a restore by creating a fake executable and a fake `project.assets.json` which only stores the relative path to the fake executable.

The mock tool package then reads the fake `project.assets.json` file to determine the executable path to its command, like the actual tool package instance would.